### DISCUSSION TITLE
GitLens would sometimes push to main branch instead of working branch.

### DISCUSSION THREAD

**Discussion by irrationnelle** [gitkraken/vscode-gitlens#3039] [Q&A]
Hello. Thanks to your help, I am effectively using GitLens in VS Code.
However, I'm encountering an issue. When I switch branches to modify my work and then push the commit, it often gets pushed to the main branch instead of the one I was working on.
This seems like a significant problem, so I believe it's not a bug, but rather an error on my part. Is there a setting I need to adjust or something specific I should check before pushing the commit?

FYI: When this problem arises, even if I switch branches to 'main' in the GitLens commit graph, the changes are not reflected and the issue persists in the current branch. I'm wondering if there's any connection.

**Comment by ramin-t** [comment]
Hi @irrationnelle,

To help understand this issue a bit better, would you mind telling me if I'm capturing these steps correctly?
1. Create a local branch from the local `main` branch
2. Switch to that branch
3. Make some changes on that branch
4. Commit the changes
5. The changes are committed to the `main` branch instead

Is that accurately describing what is happening in your case? If not, would you mind posting the issue you are seeing step by step? You mentioned changes being "pushed", so I'm wondering if remote branches might be involved, in which case we can confirm things like branch upstreams and git config settings.

**Reply by irrationnelle** [reply]
Thank you for your feedback. I'll need some time to respond as I only became aware of the issue upon pushing to the remote repository. Had I noticed that a commit was made on the main branch, I would have switched branches and performed a merge or cherry-pick then.

On a related note, I recently faced a similar issue. In the graphical Git interface at the bottom, clicking the push button didn't initiate the push, even though the button displayed the number and upward arrow, indicating unsynchronized local commits.

However, when I used the push command from the left sidebar in Visual Studio Code, the commit was successfully pushed **to the main branch instead of the working branch.**

Based on this experience, I plan to attempt another push using the bottom Git graph menu. If it fails, I'll investigate the branch and remote upstream settings at that time.

Thanks again.
